src/exporter/DaemonMetricCollector.cc

```diff
@@ -326,18 +325,6 @@ DaemonMetricCollector::get_labels_and_metric_name(std::string daemon_name,
   } else {
     labels.insert({"ceph_daemon", quote(daemon_name)});
   }
-  if (daemon_name.find("rbd-mirror") != std::string::npos) {
-    std::regex re(
-        "^rbd_mirror_image_([^/]+)/(?:(?:([^/]+)/"
-        ")?)(.*)\\.(replay(?:_bytes|_latency)?)$");
-    std::smatch match;
-    if (std::regex_search(daemon_name, match, re) == true) {
-      new_metric_name = "ceph_rbd_mirror_image_" + match.str(4);
```

Comment:
Nothing else sets `new_metric_name`, so `get_labels_and_metric_name()` can be changed to return only `labels_t` instead of a pair now.  I would also rename it to something like `get_extra_labels()` for clarity.

Reply:
This appears to be a cleanup and refactoring task related to the exporter. Would it be feasible to treat this as a distinct cleanup task? Also given we need to get this fix in ASAP

Reply:
This is a trivial refactor that should take a few minutes, so I don't see the point of delaying it.  Just make this PR consist of two commits: one for the original fix and one for the removal of the regex branch + change of signature + rename.

Reply:
Done